options.go

```diff
@@ -187,6 +192,7 @@ func ParseURL(redisURL string) (*Options, error) {
 	}
 
 	if u.User != nil {
+		o.Username = u.User.Username()
```

Comment:
As an aside I’m not sure this functionality ever worked for pre-6 with url.Parse as a plain schema://username:password@host:port/path if only one is given it is assumed to be username not password.

Reply:
ah, the docs declared `:` prefix, so this would work, resolving